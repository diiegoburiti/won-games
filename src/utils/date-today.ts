export const dateToday = new Date().toISOString().slice(0, 10)
