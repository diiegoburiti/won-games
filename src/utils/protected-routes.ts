import { getSession } from 'next-auth/client'
import { GetServerSidePropsContext } from 'next'

async function protectedRoutes(context: GetServerSidePropsContext) {
  const session = await getSession(context)

  if (!session) {
    context.res.writeHead(302, {
      Location: `/sign-in?callbackUrl=${context.resolvedUrl}`
    })
    context.res.end()
  }

  return session
}

export default protectedRoutes
