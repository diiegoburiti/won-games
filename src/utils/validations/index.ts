import Joi from 'joi'
import { UsersPermissionsRegisterInput } from 'graphql/generated/globalTypes'

const fieldsValidations = {
  username: Joi.string().min(5).required(),
  email: Joi.string()
    .email({ tlds: { allow: false } })
    .required(),
  password: Joi.string().required(),
  confirm_password: Joi.string()
    .valid(Joi.ref('password'))
    .required()
    .messages({ 'any.only': 'confirm password does not match with password' })
}

export type FieldErrors = {
  [key: string]: string
}

function getFieldErros(objError: Joi.ValidationResult) {
  const errors: FieldErrors = {}

  if (objError.error) {
    objError.error.details.forEach((err) => {
      errors[err.path.join('.')] = err.message
    })
  }

  return errors
}

type signInValues = Omit<UsersPermissionsRegisterInput, 'username'>

export function signInValidate(values: signInValues) {
  const { email, password } = fieldsValidations
  const schema = Joi.object({ email, password })
  const fieldErros = schema.validate(values, { abortEarly: false })
  return getFieldErros(fieldErros)
}

export function signUpValidate(values: UsersPermissionsRegisterInput) {
  const schema = Joi.object(fieldsValidations)
  const fieldErros = schema.validate(values, { abortEarly: false })
  return getFieldErros(fieldErros)
}
