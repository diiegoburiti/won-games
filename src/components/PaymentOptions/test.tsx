import { screen, waitFor } from '@testing-library/react'
import paymentMock from './mock'

import PaymentOptions from '.'
import { render } from 'utils/test-utils'

import userEvent from '@testing-library/user-event'

describe('<PaymentOptions />', () => {
  it('should render the heading', () => {
    const { container } = render(
      <PaymentOptions cards={paymentMock} handlePayment={jest.fn} />
    )

    expect(
      screen.getByRole('heading', { name: /Payment/i })
    ).toBeInTheDocument()

    expect(screen.getByRole('img', { name: /mastercard/i })).toBeInTheDocument()
    expect(screen.getByRole('img', { name: /visa/i })).toBeInTheDocument()
    expect(
      screen.getByRole('button', { name: /add a new credit card/i })
    ).toBeInTheDocument()
    expect(screen.getByLabelText(/4325/)).toBeInTheDocument()
    expect(screen.getByLabelText(/4326/)).toBeInTheDocument()

    expect(container.firstChild).toMatchSnapshot()
  })

  it('should handle select card when clicking on the label', async () => {
    render(<PaymentOptions cards={paymentMock} handlePayment={jest.fn} />)

    userEvent.click(screen.getByRole('radio', { name: /4325/i }))

    await waitFor(() => {
      expect(screen.getByRole('radio', { name: /4325/i })).toBeChecked()
    })
  })

  it('should not call handlePayment when button is disabled', async () => {
    const handlePayment = jest.fn()
    render(<PaymentOptions cards={paymentMock} handlePayment={handlePayment} />)

    userEvent.click(screen.getByRole('button', { name: /buy now/i }))
    expect(handlePayment).not.toBeCalled()
  })

  it('should not call handlePayment when button is disabled', async () => {
    const handlePayment = jest.fn()
    render(<PaymentOptions cards={paymentMock} handlePayment={handlePayment} />)

    userEvent.click(screen.getByRole('radio', { name: /4325/i }))
    userEvent.click(screen.getByRole('button', { name: /buy now/i }))

    await waitFor(() => {
      expect(handlePayment).toHaveBeenCalled()
    })
  })
})
