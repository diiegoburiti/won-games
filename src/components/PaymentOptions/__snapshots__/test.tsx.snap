// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PaymentOptions /> should render the heading 1`] = `
.c8 {
  display: inline-block;
  vertical-align: middle;
  overflow: hidden;
}

.c11 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  background: linear-gradient(180deg,#ff5f5f 0%,#f062c0 50%);
  color: #FAFAFA;
  border: 0;
  border-radius: 0.4rem;
  padding: 0.8rem;
  cursor: pointer;
  -webkit-text-decoration: none;
  text-decoration: none;
  height: 4rem;
  font-size: 1.4rem;
  padding: 0.8rem 3.2rem;
  width: 100%;
  background: none;
  color: #F231A5;
}

.c11:hover {
  background: none;
}

.c11:hover {
  color: #e20e8d;
}

.c12 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  background: linear-gradient(180deg,#ff5f5f 0%,#f062c0 50%);
  color: #FAFAFA;
  border: 0;
  border-radius: 0.4rem;
  padding: 0.8rem;
  cursor: pointer;
  -webkit-text-decoration: none;
  text-decoration: none;
  height: 4rem;
  font-size: 1.4rem;
  padding: 0.8rem 3.2rem;
  width: 100%;
}

.c12:hover {
  background: linear-gradient(180deg,#e35565 0%,#d958a6 50%);
}

.c12 svg {
  width: 1.5rem;
}

.c12 svg + span {
  margin-left: 0.8rem;
}

.c12:disabled {
  cursor: not-allowed;
  -webkit-filter: saturate(40%);
  filter: saturate(40%);
}

.c1 {
  color: #030517;
  position: relative;
  margin-bottom: 3.2rem;
  font-size: 1.6rem;
}

.c1::after {
  position: absolute;
  left: 0;
  bottom: -0.5rem;
  content: '';
  width: 5rem;
  border-bottom: 0.5rem solid #F231A5;
}

.c1::after {
  width: 3rem;
}

.c5 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c6 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
  width: 1.8rem;
  height: 1.8rem;
  border: 0.2rem solid #F231A5;
  border-radius: 50%;
  background: transparent;
  -webkit-transition: background 0.1s ease-in-out;
  transition: background 0.1s ease-in-out;
  outline: none;
  cursor: pointer;
}

.c6:focus {
  box-shadow: 0 0 0.5rem #F231A5;
}

.c6:before {
  content: '';
  width: 0.8rem;
  height: 0.8rem;
  border-radius: 50%;
  background: #F231A5;
  -webkit-transition: opacity 0.1s ease-in-out;
  transition: opacity 0.1s ease-in-out;
  opacity: 0;
  position: absolute;
}

.c6:checked:before {
  opacity: 1;
}

.c0 {
  padding: 2.4rem;
  background: #FAFAFA;
}

.c9 {
  background: #eee;
  color: #030517;
  font-weight: 600;
  padding: 2.4rem;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c9 .c10 {
  padding-left: 0.8rem;
  padding-right: 0.8rem;
  outline: 0;
}

.c2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

.c4 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c4 img {
  margin-right: 0.8rem;
}

.c3 {
  background: #EAEAEA;
  border-radius: 0.2rem;
  color: #030517;
  padding: 0 0.8rem;
  height: 5rem;
  cursor: pointer;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
}

.c3:not(:last-child) {
  margin-bottom: 0.8rem;
}

.c7 {
  background: #EAEAEA;
  border-radius: 0.2rem;
  color: #030517;
  padding: 0 0.8rem;
  height: 5rem;
  cursor: pointer;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c7 svg {
  margin-left: 0.8rem;
  margin-right: 1.6rem;
  width: 2.4rem;
}

<div
  class=""
>
  <div
    class="c0"
  >
    <h2
      class="c1"
      color="black"
    >
      Payment
    </h2>
    <div
      class="c2"
    >
      <label
        class="c3"
      >
        <span
          class="c4"
        >
          <img
            alt="visa"
            src="/img/visa.png"
          />
          *** *** **** 4325
        </span>
        <div
          class="c5"
        >
          <input
            class="c6"
            id="*** *** **** 4325"
            name="credit-card"
            type="radio"
            value="*** *** **** 4325"
          />
        </div>
      </label>
      <label
        class="c3"
      >
        <span
          class="c4"
        >
          <img
            alt="mastercard"
            src="/img/master-card.png"
          />
          *** *** **** 4326
        </span>
        <div
          class="c5"
        >
          <input
            class="c6"
            id="*** *** **** 4326"
            name="credit-card"
            type="radio"
            value="*** *** **** 4326"
          />
        </div>
      </label>
      <div
        aria-label="add a new credit card"
        class="c7"
        role="button"
      >
        <svg
          aria-hidden="true"
          class="c8"
          fill="currentColor"
          focusable="false"
          height="14"
          viewBox="0 0 24 24"
          width="14"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M0 0h24v24H0V0z"
            fill="none"
          />
          <path
            d="M19 13h-6v6h-2v-6H5v-2h6V5h2v6h6v2z"
          />
        </svg>
         Add a new credit card
      </div>
    </div>
  </div>
  <div
    class="c9"
  >
    <a
      class="c10 c11"
    >
      <span>
        Continue shopping
      </span>
    </a>
    <button
      class="c10 c12"
      disabled=""
    >
      <svg
        aria-hidden="true"
        class="c8"
        fill="currentColor"
        focusable="false"
        viewBox="0 0 24 24"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M0 0h24v24H0V0z"
          fill="none"
        />
        <path
          d="M15.55 13c.75 0 1.41-.41 1.75-1.03l3.58-6.49A.996.996 0 0020.01 4H5.21l-.94-2H1v2h2l3.6 7.59-1.35 2.44C4.52 15.37 5.48 17 7 17h12v-2H7l1.1-2h7.45zM6.16 6h12.15l-2.76 5H8.53L6.16 6zM7 18c-1.1 0-1.99.9-1.99 2S5.9 22 7 22s2-.9 2-2-.9-2-2-2zm10 0c-1.1 0-1.99.9-1.99 2s.89 2 1.99 2 2-.9 2-2-.9-2-2-2z"
        />
      </svg>
      <span>
        Buy now
      </span>
    </button>
  </div>
</div>
`
